src/app/data-model-provider/MetadataLookup.cpp

```diff
@@ -63,23 +65,26 @@ std::optional<AttributeEntry> AttributeFinder::Find(const ConcreteAttributePath
     return std::nullopt;
 }
 
-EndpointFinder::EndpointFinder(ProviderMetadataTree * provider) : mProvider(provider)
+Protocols::InteractionModel::Status ValidateClusterPath(ProviderMetadataTree * provider, const ConcreteClusterPath & path,
+                                                        Protocols::InteractionModel::Status successStatus)
 {
-    VerifyOrReturn(mProvider != nullptr);
-    mEndpoints = mProvider->EndpointsIgnoreError();
-}
+    if (ServerClusterFinder(provider).Find(path).has_value())
+    {
+        return successStatus;
+    }
 
-std::optional<EndpointEntry> EndpointFinder::Find(EndpointId endpointId)
-{
-    for (auto & endpointEntry : mEndpoints)
+    // if we get here, the path is invalid.
```

Comment:
```suggestion
    // If we get here, the cluster identified by the path does not exist.
```
The path is _valid_ as far as that goes.  It just points to nothing.